cdap-data-fabric/src/main/java/io/cdap/cdap/spi/metadata/dataset/SearchHelper.java

```diff
@@ -304,22 +313,62 @@ private MetadataSearchResponse search(Set<MetadataScope> scopes, SearchRequest r
       finalResults, cursors, request.shouldShowHidden(), request.getEntityScopes());
   }
 
-  private Set<MetadataEntity> getSortedEntities(List<MetadataEntry> results, SortInfo sortInfo) {
+  private Map<MetadataEntity, Map<String, Integer>> hashResults(List<MetadataResultEntry> results) {
+    Map<MetadataEntity, Map<String, Integer>> hashedResults = new LinkedHashMap<>();
+    for (MetadataResultEntry resultEntry : results) {
+      MetadataEntity entity = resultEntry.getMetadataEntity();
+      String searchTerm = resultEntry.getSearchTerm();
+      Map<String, Integer> map = hashedResults.getOrDefault(entity, new LinkedHashMap<>());
+      map.put(searchTerm, map.getOrDefault(searchTerm, 0) + 1);
+      hashedResults.putIfAbsent(entity, map);
```

Comment:
Seems like a waster to compute the `map` but then it might not be used. We can check if the `hashedResults` contains the `entity` first? Or you can use `computeIfAbsent`.

Reply:
Hm, so I had used `.putIfAbsent` but I actually think that this line always puts something, and so `.put` would be more appropriate 

because right now I cannot draw out a situation where it would be absent..

Reply:
I looked at the logic again, and I think you should do it like this:

```java
Map<String, Integer> map = hashedResults.computeIfAbsent(entity, k -> new LinkedHashMap<>());
map.put(...);
```
That's it and you don't need to call `hashedResults.put` at all.

Reply:
that works, thanks!